pages/api/v1/verification/index.ts

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019-2025 @polkassembly/polkassembly authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import type { NextApiHandler } from 'next';
+import sgMail from '@sendgrid/mail';
+import withErrorHandling from '~src/api-middlewares/withErrorHandling';
+import { MessageType } from '~src/auth/types';
+import firebaseAdmin from '~src/services/firebaseInit';
+import { cryptoRandomStringAsync } from 'crypto-random-string';
+import getTokenFromReq from '~src/auth/utils/getTokenFromReq';
+import authServiceInstance from '~src/auth/auth';
+import apiErrorWithStatusCode from '~src/util/apiErrorWithStatusCode';
+import messages from '~src/auth/utils/messages';
+
+interface IReq {
+	type: 'email' | 'twitter',
+	checkingVerified?:boolean;
+}
+
+export enum VerificationStatus {
+	ALREADY_VERIFIED = 'Already verified',
+	VERFICATION_EMAIL_SENT = 'Verification email sent',
+	PLEASE_VERIFY_TWITTER = 'Please verify twitter',
+	NOT_VERIFIED = 'Not verified'
+}
+
+export interface IVerificationResponse {
+	status: VerificationStatus;
+}
+
+const apiKey = process.env.SENDGRID_API_KEY;
+const FROM = {
+	email: 'noreply@polkassembly.io',
+	name: 'Polkassembly'
+};
+
+if (apiKey) {
+	sgMail.setApiKey(apiKey);
+}
+
+const firestore = firebaseAdmin.firestore();
+
+const handler: NextApiHandler<IVerificationResponse | MessageType> = async (req, res) => {
+	const { type, checkingVerified } = req.query as unknown as IReq;
```

Comment:
why do we have data accepted in both query and params ?
also if you require data in `req.body` implement an early check for request type to be POST only